docs/design/virtual-addresses.md

```diff
@@ -0,0 +1,155 @@
+# Virtual Addresses
+
+## Purpose
+Introduce the concept of virtual addresses in order to bridge the conceptual gap between Hedera's and Ethereum's view of accounts.
+Virtual addresses will resolve the issue of account EVM compatibility and identification and will be responsible for EVM address related logic.
+
+## Goals
+- Add list of evmAddress values known as “Virtual Address” to Hedera accounts which govern the address the EVM observes for a given account transaction
+- Hedera Accounts can add, disable and remove virtual address entries as desired
+- The Hedera network will:
+  - Validate ownership by extracting the public key from the signature and comparing the calculated public address to the `evmAddress` passed in on addition of the virtual address
+  - Maintain an `evmAddress -> accountId` map thereafter
+- Contract accounts may utilize the `evmAddress` to store their address in accordance with `CREATE` and `CREATE2` EVM operations
+- Restore HIP 32's consistency
+  - The values of the account aliases will only contain EC/ED key bytes (as in HIP 32)
+  - The values of the account aliases will no longer contain EVM addresses
+- Perform seamless state migration of accounts and contracts so that by default they benefit from the introduction of virtual addresses
+
+## Non Goals
+- Support removal of virtual addresses between accounts
+- Support transferring of virtual addresses between accounts
+
+## Architecture
+
+### Account state updates
+- Update `HederaAccount` interface to have the ability to:
+  - Read/write virtual addresses
+  - Read/write a default virtual address
+  - Track nonce values for each virtual address
+```
+public interface HederaAccount {
+...
+
+	Set<ByteString> getVirtualAddresses();
+
+	void setVirtualAddresses(Set<ByteString> virtualAddresses);
```

Comment:
Why do we need `setVirtualAddresses(Set<ByteString> virtualAddresses)`?  Will not getting the set via `getVirtualAddresses()`suffice to allow us to manipulate the contents of the set (add, delete)?

Also, how will we mark a virtual address as disabled?

Reply:
> Why do we need setVirtualAddresses(Set<ByteString> virtualAddresses)? Will not getting the set via getVirtualAddresses()suffice to allow us to manipulate the contents of the set (add, delete)?

I think the idea was to work with immutable collection instead of mutating an underlying one.

> Also, how will we mark a virtual address as disabled?

Hmm, good point. We can extend the map field that we were using for nonces to have values of something like `VirtualAddressSpec`. @Nana-EC any thoughts?